### PR TITLE
ENH+TST: cfg_yoda will commit selectively and fail on dirty

### LINKED ISSUE #3666: `cfg_yoda` saves any untracked content
Found while composing https://github.com/datalad-handbook/book/pull/158#discussion_r324167794

The procedure should selectively save the pieces it touched.

In general I wonder, whether it is a good idea to have `run-procedure` operate on dirty datasets (by default). It makes it impossible to tell which changes are really coming from the procedure, unless procedure authors (which could be anyone) are mindful of this setup.

### PR DESCRIPTION
**Pull request by mih**
Before it would save the entire state of a dataset. Fixes #3666

Adjust and extend tests to cover the new behavior.

### CODE REVIEW

**Comment by mih** [comment]
Unreleated appveyor failure.

File: datalad/resources/procedures/cfg_yoda.py [hunk 1]
```diff
@@ -15,6 +15,30 @@
...
 
+to_modify = [
+    ds.pathobj / 'code' / 'README.md',
+    ds.pathobj / 'code' / '.gitattributes',
+    ds.pathobj / 'README.md',
+    ds.pathobj / 'CHANGELOG.md',
+    ds.pathobj / '.gitattributes',
+]
+
+dirty = [
```
Comment (yarikoptic) [inline datalad/resources/procedures/cfg_yoda.py]: here it probably doesn't matter either way, but IMHO better to use generators more to gain some efficiency.  E.g. making here it into a `()`, and below condition into `if any(dirty):`

Comment (mih) [reply datalad/resources/procedures/cfg_yoda.py]: Hmm, but wouldn't the generator be consumed by `any()` and leave me no info to put in the exception?

**Comment by mih** [comment]
OK, let's move.

File: datalad/resources/procedures/cfg_yoda.py (hunk 1 above)
Comment (yarikoptic) [reply datalad/resources/procedures/cfg_yoda.py]: ah, right. sorry I have missed that it was used below!  Sorry for the noise, disregard original comment ;)

**Pull request merged**